move_base/include/move_base/move_base.h

```diff
@@ -76,6 +76,40 @@ namespace move_base {
     OSCILLATION_R
   };
 
+  //Used to calculate distance, time and average velocity
+  struct FeedbackInfo
+  {
+      FeedbackInfo()
+      {
+        clear();
+      }
+
+      void clear()
+      {
+        velocity = MIN_VELOCITY;
+        is_set = false;
+      }
+
+      double getTime(double distance, double velocity)
+      {
+        double epsilon = 0.01;
+        if(fabs(velocity - 0.0) < epsilon)
+        {
+          velocity = MIN_VELOCITY;
+        }
+
+        return fabs(distance/velocity);
+      }
+
+      double dist_to_goal;
+      double velocity;
+      ros::Time time;
+      bool is_set;
+      //the value between(0,1)
+      static constexpr double WEIGHT_AVERAGE_VELOCITY_FACTOR = 0.9;
```

Comment:
This factor corresponds roughly to averaging over 10 iterations (1/(1-0.9)) which corresponds to <1 sec (controller runs at ~13hz).
I think we need more, 5-10 sec. Maybe even worth making this param configurable.